### ISSUE TITLE
Memory-Leak with text in links example

### ISSUE THREAD

**Issue by Lowizer** [vasturiano/force-graph#140]
With your example [Text in links](https://github.com/vasturiano/force-graph/blob/master/example/text-links/index.html) i get a huge memory leak. In a few minutes my RAM gets filled with gigabytes of supposedly unused data. Every so often the garbage collector clears some memory but can't keep up until my RAM gets filled completely.

1. Open the example mentioned above.
2. Open the task manager of your browser.
3. Observe the memory usage going up.

Expected behavior:
No more leaks.

Desktop:
 - OS: Tested on Manjaro Linux, Linux Mint and Arch Linux
 - Browser: Firefox, Chrome, Vivaldi

Additional context
Copied your code into my own project and as soon as i display some strings on the links i get a huge memory leak.

**Comment by Alexithemia** [comment]
Same here, I think It has something to do with the aligning the text with the link, it does a lot of processing. If you take out the math to align the text it works fine for me.

**Comment by vasturiano** [comment]
Yes, the biggest burden in that example is actually the rotation of the text to follow the link line:
https://github.com/vasturiano/force-graph/blob/2717185a8c58d362fc7654061b8dab07622c5137/example/text-links/index.html#L56

If you remove that step, the performance should improve dramatically.

I suppose the canvas rotation is an expensive operation as it needs to re-calculate the transformation matrixes. Possibly one way to improve it is to set the transformations manually using `setTransform`, but that's a more involved math for a simple example.
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/setTransform
